commands_and_menus/README.md

```diff
@@ -0,0 +1,180 @@
+# Commands and Menus 
+
+* [Jupyterlab Commands](#jupyterlab-commands)
+* [Adding new Menu tabs and items](#adding-new-menu-tabs-and-items)
+
+## Commands and Menus: Extending the main app
+
+For the next extension you can either copy the last folder to a new one or
+simply continue modifying it. In case that you want to have a new extension,
+open the file `package.json` and modify the package name, e.g. into
+`2_commands_and_menus`. The same name changes needs to be done in
+`src/index.ts`.
+
+#### Jupyterlab Commands ####
+
+Start it with `jupyter lab --watch`. In this extension, we are going to add a
+command to the application command registry and expose it to the user in the
+command palette.  The command palette can be seen when clicking on _Commands_
+on the left hand side of Jupyterlab. The command palette can be seen as a list
+of actions that can be executed by JupyterLab. (see screenshot below).
+
+![Jupyter Command Registry](_images/command_registry.png)
+
+Extensions can provide a bunch of functions to the JupyterLab command registry
+and then expose them to the user through the command palette or through a menu
+item.
+
+Two types play a role in this: the `CommandRegistry` type ([documentation](http://phosphorjs.github.io/phosphor/api/commands/classes/commandregistry.html))
+and the command palette interface `ICommandPalette` that is imported with:
+
+```typescript
+import {
+  ICommandPalette
+} from '@JupyterLab/apputils';
+```
+
+To see how we access the applications command registry and command palette
+open the file `src/index.ts`.
+
+```typescript
+const extension: JupyterLabPlugin<void> = {
+    id: '2_commands_and_menus',
```

Comment:
Maybe we can also remove the numbers from the extension ids?

Reply:
I have removed them and done a quick pass for better format. I think it is in a state where it can be merged to then go with dedicated PR for each of the examples to polish and make them better.